### DISCUSSION TITLE
Typescript: Warning: An update to MyComponent inside a test was not wrapped in act(...)

### DISCUSSION THREAD

**Discussion by JohanAltamar** [react-hook-form/react-hook-form#7125] [Q&A] [answered]
Good morning guys, I'm facing this warning while testing but when implemented the recommended solution I'm seeing the following TS errors, do you have any idea about how to solve them? I'll appreciate them.
```
No overload matches this call.
  Overload 1 of 2, '(callback: () => Promise<void>): Promise<undefined>', gave the following error.
    Type 'Promise<RenderResult<typeof xxx), HTMLElement>>' is not assignable to type 'Promise<void>'.
    Type 'RenderResult<typeof xxx), HTMLElement>' is not assignable to type 'void'.
  Overload 2 of 2, '(callback: () => VoidOrUndefinedOnly): void', gave the following error.
    Type 'Promise<RenderResult<typeof xxx), HTMLElement>>' is not assignable to type 'VoidOrUndefinedOnly'.
    Property '[UNDEFINED_VOID_ONLY]' is missing in type 'Promise<RenderResult<typeof xxx), HTMLElement>>' but required in type '{ [UNDEFINED_VOID_ONLY]: never; }'.ts(2769)

```

**Comment by nikitastryuk** [comment, marked as answer]
Had same error
You see this because `act` method return type (`Promise<void>`) is not assignable to `render` one (which is not `Promise<void>`).
<img width="1269" alt="image" src="https://user-images.githubusercontent.com/32720808/154056886-32a263e0-d5e0-4b5a-b1fd-15549b196615.png">

Solution could be to make it return `Promise<void>` like this:
<img width="1039" alt="image" src="https://user-images.githubusercontent.com/32720808/154056953-0750b252-caf7-47e5-9d2a-c64a7a713f19.png">

**Reply by Moshyfawn** [reply]
Fair point!
@bluebill1049 do you want me update the docs to include
```js
await act(async () => {
  render(<App />);
});
```
instead of
```js
await act(async () => render(<App />));
```

It might help to avoid some confusion as many devs often copy-paste examples assuming it would just work

**Reply by bluebill1049** [reply]
thank you @Moshyfawn 🙏 perhaps we can have it in this section: https://react-hook-form.com/advanced-usage#TestingForm

**Reply by Moshyfawn** [reply]
There's already an example of resolving the act warning during test.

![image](https://user-images.githubusercontent.com/16290753/154060472-75a060b7-178c-41fb-8228-e9377bb43127.png)

It's just a matter of updating the example to not return the result of `render()` call. Sounds good?

**Reply by bluebill1049** [reply]
sounds great! thanks @Moshyfawn 🙏

**Reply by nikitastryuk** [reply]
thank you, guys

**Reply by Moshyfawn** [reply]
Thank you for looking into the matter, @nikitastryuk!

**Comment by Moshyfawn** [comment]
Y'all can't think it's a very helpful way to ask for support, right?
Please, consider providing an isolated codesanbox reproduction instead

**Comment by nikitastryuk** [comment]
Investigated the issue and updated my comment above. 
I think it would nice to make docs use my example to avoid possible confusion.
